### PR TITLE
wasm2c: build library wasm-rt-impl

### PR DESCRIPTION
**Pull request by eadwu**
Also include header files needed for compilation

### CODE REVIEW

**Comment by eadwu** [comment]
I'm guessing this is failing because `CMAKE_INSTALL_*` isn't defined. I'll figure out how this works.

**Comment by binji** [comment]
Cool, seems like a good idea!

File: CMakeLists.txt [hunk 1]
```diff
@@ -324,6 +325,10 @@ set(WABT_LIBRARY_SRC
 
 add_library(wabt STATIC ${WABT_LIBRARY_SRC})
 
+add_library(wasm-rt-impl STATIC wasm2c/wasm-rt-impl.c wasm2c/wasm-rt-impl.h)
+INSTALL(TARGETS wasm-rt-impl DESTINATION ${CMAKE_INSTALL_LIBDIR})
+INSTALL(FILES wasm2c/wasm-rt.h wasm2c/wasm-rt-impl.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```
Comment (sbc100) [inline CMakeLists.txt:330]: The convention in this file is to use lowercase function names

File: CMakeLists.txt [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 cmake_minimum_required(VERSION 3.0.0)
 project(WABT VERSION 1.0.13)
+include(GNUInstallDirs)
```
Comment (sbc100) [inline CMakeLists.txt:19]: Why is these needed?

Comment (eadwu) [reply CMakeLists.txt:19]: `CMAKE_INSTALL_LIBDIR` isn't set without by default without them which would fail the install, not sure where the default location to install them is without it.

Not sure about the compatibility between OS outside of Linux either.

Binaries installed to `bin/` seem to be done manually here and I can try adapting that over to fit with `lib/`?

File: CMakeLists.txt (hunk 1 above)
Comment (eadwu) [reply CMakeLists.txt]: Adjusted.

**Comment by nomeata** [comment]
I noticed that nixpkgs is applying this patch. Would be nice to bless it official?

**Comment by sbc100** [comment]
The patch is approved.. but it looks like it needs a rebase before we can land.

**Comment by eadwu** [comment]
Rebased.

**Comment by sbc100** [comment]
Looks like a real windows build failure.

**Comment by binji** [comment]
Yep, wasm2c still isn't quite right on Windows (https://github.com/WebAssembly/wabt/tree/wasm2c-windows tries to fix, but is incomplete). So this library shouldn't be built there.

**Comment by eadwu** [comment]
Fixed.

**Comment by binji** [comment]
Thanks!
